pkg/datapath/linux/probes/probes.go

```diff
@@ -201,53 +201,25 @@ func (p *ProbeManager) SystemKernelHz() (int, error) {
 // returns an error when parameters required by Cilium are not enabled. It logs
 // warnings when optional parameters are not enabled.
 func (p *ProbeManager) SystemConfigProbes() error {
-	config := p.features.SystemConfig
-
-	if !p.KernelConfigAvailable() {
-		return ErrKernelConfigNotFound
-	}
-
-	// Required
-	if !config.ConfigBpf.Enabled() {
-		return fmt.Errorf("CONFIG_BPF kernel parameter is required")
-	}
-
-	if !config.ConfigBpfSyscall.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_BPF_SYSCALL kernel parameter is required")
-	}
-	if !config.ConfigNetSchIngress.Enabled() && !config.ConfigNetSchIngress.Module() {
-		return fmt.Errorf(
-			"CONFIG_NET_SCH_INGRESS kernel parameter (or module) is required")
-	}
-	if !config.ConfigNetClsBpf.Enabled() && !config.ConfigNetClsBpf.Module() {
-		return fmt.Errorf(
-			"CONFIG_NET_CLS_BPF kernel parameter (or module) is required")
-	}
-	if !config.ConfigNetClsAct.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_NET_CLS_ACT kernel parameter is required")
-	}
-	if !config.ConfigBpfJit.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_BPF_JIT kernel parameter is required")
-	}
-	if !config.ConfigHaveEbpfJit.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_HAVE_EBPF_JIT kernel parameter is required")
-	}
-	// Optional
-	if !config.ConfigCgroupBpf.Enabled() {
-		log.Warning(
-			"CONFIG_CGROUP_BPF optional kernel parameter is not in kernel configuration")
+	requiredParams := p.GetRequiredConfig()
+	for param, status := range requiredParams {
+		var canBeModule string
+		switch param {
+		case "CONFIG_NET_SCH_INGRESS", "CONFIG_NET_CLS_BPF":
+			canBeModule = " (or module)"
+		}
+		if !status {
+			return fmt.Errorf("%s kernel parameter%s is required", param, canBeModule)
+		}
 	}
-	if !config.ConfigLwtunnelBpf.Enabled() {
-		log.Warning(
-			"CONFIG_LWTUNNEL_BPF optional kernel parameter is not in kernel configuration")
+	optionalParams := p.GetOptionalConfig()
+	for param, status := range optionalParams {
+		if !status {
+			log.Warning("%s kernel parameter is optional", param)
```

Comment:
Please keep the original message, `"%s optional kernel parameter is not in kernel"`.

Reply:
fixed.